### PR TITLE
Allow signed profiles to be uploaded by checking for PKCS7

### PR DESCRIPTION
**Pull request by jessepeterson**

### CODE REVIEW

File: profile/profile.go [hunk 1]
```diff
@@ -16,9 +17,22 @@ type payloadIdentifier struct {
 }
 
 func (mc *Mobileconfig) GetPayloadIdentifier() (string, error) {
+	mcBytes := *mc
+	if len(mcBytes) > 5 && string(mcBytes[0:5]) != "<?xml" {
+		p7, err := pkcs7.Parse(mcBytes)
+		if err != nil {
+			return "", errors.Wrapf(err, "Mobileconfig is not XML nor PKCS7 parseable")
+		}
+		err = p7.Verify()
```
Comment (groob) [inline profile/profile.go]: what CA does the call to Verify use?

File: profile/profile.go [hunk 2]
```diff
@@ -16,9 +17,22 @@ type payloadIdentifier struct {
...
+		if err != nil {
+			return "", err
+		}
+		mcBytes = Mobileconfig(p7.Content)
+	}
+
 	// TODO: support CMS signed profiles
```
Comment (groob) [inline profile/profile.go]: remove the TODO?

File: profile/profile.go (hunk 1 above)
Comment (jessepeterson) [inline profile/profile.go]: It doesn't verify any certificates. PKCS7 verify merely verifies the signature of the data is valid.

Comment (jessepeterson) [inline profile/profile.go]: https://github.com/fullsailor/pkcs7/blob/master/pkcs7.go#L208-L221

Comment (jessepeterson) [inline profile/profile.go]: This is similar in the OpenSSL world, too — verifying the signature cryptographically verifies against the data (but not the included certificates, or where they're from, or anything like that).

File: profile/profile.go (hunk 2 above)
Comment (jessepeterson) [inline profile/profile.go]: oh lord. yes, sorry :)

File: profile/profile.go (hunk 1 above)
Comment (groob) [inline profile/profile.go]: Makes sense. Thanks for the refresher on how this works:)

**Pull request merged**